src/Auth0LoginProcessor.ts

```diff
@@ -41,15 +34,15 @@ type Config = {
  * At the end of the authentication process, that did not time out, it might take a couple of seconds
  * for the NodeJS HTTP server to close.
  */
-export
-class Auth0LoginProcessor {
+export class Auth0LoginProcessor<TToken> {
   private readonly server: http.Server = http.createServer(this.handleAuth0Response.bind(this))
   private csrfToken: string = ''
   private codeVerifier: string = ''
   private authResponse: Deferred<AuthResponse> = mkDeferred()
-  
+
   constructor(
-    public readonly config: Config
+    public readonly config: Config,
```

Comment:
I assume that this validation is wanting to be `TypeValidator` but we don't have a public library for that yet.
Still, the fact that the constructor constructs a bit but validates a-lot - hints to me that the validation is ought to be a function over `Config`, that the constructor merely calls

Reply:
Sounds good. But, this PR is about a different change.
Feel free to submit a new PR.